python/ray/serve/pipeline/deployment_node.py

```diff
@@ -194,7 +199,10 @@ def __getattr__(self, method_name: str):
             (),
             {},
             {},
-            other_args_to_resolve=self._bound_other_args_to_resolve,
+            other_args_to_resolve={
+                **self._bound_other_args_to_resolve,
+                PARENT_CLASS_NODE_KEY: self,
```

Comment:
used by unit test, not by the actual ray dag -> serve dag translation path

Reply:
we're only passing it so `DeploymentMethodNode` can find `method_body = getattr(self._deployment_node, self._deployment_method_name)` that's essentially a deployment handle method call, do we need to include the actual instance of `DeploymentNode` in `other_args_to_resolve` ?

Given `other_args_to_resolve` is on critical path of DAG execution and json serde it's preferable if we add minimal info and avoid passing in `Deployment` or `DAGNode` objects if possible.